packages/gatsby-plugin-fastify/src/utils/routes.ts

```diff
@@ -12,6 +12,7 @@ export function removeQueryParmsFromUrl(url: string) {
 
 export function buildUrlFromParams(path: string, data: { [s: string]: string } = {}) {
   return path.replace(/:(\w+)|(\*)/gi, function (_match, p1, p2) {
+    if (p1 && !data[p1]) return _match; // :Something in toPath does not have a splat in fromPath pass it through colon intact
```

Comment:
Also, should we be moving this into to the error function below? what do you think? I think we still have to error if wildcard routes aren't fund cause the `*` is indeed not allowed if I remember correctly. This is a nit, not to worried either way.

Reply:
Maybe I'm misunderstanding, but I believe what you're saying is that if there is a wildcard in the `toPath` but not in the `fromPath` then it should still error as before because there shouldn't be an asterisk in the `toPath`? If so, I'm not sure about whether asterisk is valid or not in the URL but it does still make it to the error in the case where it detects a wildcard in to the `toPath`.  I've added a commit with a test case for this.

Reply:
Yeah, you got that right. Sounds like the functionality we want.